networking.md

```diff
@@ -0,0 +1,463 @@
++--------------+-----------+------------------------------------------------------------------+
+```
+
+#### VIP endpoints
+MetalLB service VIP (currently - in layer 2 mode) used for endpoints on networks which compute nodes need to be able to access. The connection is terminated at one node at the metallb VIP. From there spread to all service pods via the networking rules kube-proxy has set.
+
+The OCP LoadBalancer type service name gets registered as the keystone endpoint for the service. Controlplane services running on OCP will resolve those to the ClusterIP of the service, the Compute nodes have to resolve it to the External-IP of the LoadBalancer.
+```bash
++--------------+-----------+---------------------------------------------------------------+
+| Service Name | Interface | URL                                                           |
++--------------+-----------+---------------------------------------------------------------+
+| <service>    | internal  | http://<service>-internal.openstack.svc:9292                  |
++--------------+-----------+---------------------------------------------------------------+
+```
+
+#### Individual endpoints
+For services like OVS access via static IP on the isolated network is required.
+
+This is right now a TODO and targeted when the IPAM solution, which is also required for EDPM nodes has finalized. Right now when the OVS pods restart, they'll likely get different IPs.
+
+The goal is for those pods to use static IPAM plugin and pass the pod IP via the annotation.
+```yaml
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[{"name": "internalapi", "namespace": "openstack", "ips": ["172.17.0.20/24"]}]'
+```
+
+### Egress
+For outgoing isolated network traffic, pods be attached to additional networks using multus. The operators add the `k8s.v1.cni.cncf.io/networks` annotation  with referencing the list of network-attachment-definitios the deployemtn pods should be attached:
+
+```yaml
+annotations:
+  k8s.v1.cni.cncf.io/networks: '[{"name": "internalapi", "namespace": "openstack"},{"name": "storage", "namespace": "openstack"}]'
+```
+
+## Env Pre-requisite
+The NMstate and MetalLB operator need to be deployed and the OCP environment must support Multis CNI
+
+### NMstate operator
+The NMstate operator is performing state-driven network configuration across the cluster nodes with NMState. Further details can be found at:
+
+* [About the Kubernetes NMState Operator](https://docs.openshift.com/container-platform/4.12/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html)
+* [The Kubernetes NMState project](https://nmstate.io/kubernetes-nmstate/)
+
+### MetalLB operator
+The MetalLB operator provides a load-balancer implementation for bare metal clusters which its aim is to bring user traffic into the clusters network using a load balancer implementation. Currently only tested using MetalLB layer 2 mode. Further details can be found at:
```

Comment:
The first sentence here seems redundant in its mention of `load-balancer implementation`.  Maybe it would be enough to say something like:

```
The MetalLB operator provides a load-balancer implementation for bare metal clusters which brings user traffic into the clusters network.
```